### DISCUSSION TITLE
.HEX File

### DISCUSSION THREAD

**Discussion by Faubre** [fra589/grbl-Mega-5X#180] [Q&A] [answered]
Hi,
We're building a 5 axes CNC Machine and we had some trouble to flash a .HEX file intro arduino.
How can i get my .HEX file from GRBL-MEGA-5X-1.1g....

Thanx for all 

ps: Je suis Francais en passant!

**Comment by RaphaelDives** [comment]
Hey,
come on... you're building an CNC machine, but you are not able to flash a hex file? ;) (no offence)

Although I would never ever recommend to upload a "unkown" hex file to my arduino... (especially when you need a special 5axis configuration) you can:

1) use the arduino avrdude upload tool (i think it's an commandline program)
2) use a programmer (i have the atmel-ice) together with atmel studio (recommended!)
3) any ISP programmer (paralel/serial/usb) with an upload tool. I used "PonyProg" a long, long time ago...
4) i think you can also program an arduino as programmer... ;)

Good luck!
Cheers,
Raphael

**Comment by fra589** [comment, marked as answer]
Hello @Faubre,

Tu es sous Windows (quelle version) ou sous Linux (quelle distrib/version)?
Quel est ton problème exactement ? Tu n'arrives pas à compiler ? Grbl se compile mais tu n'arrives pas à flasher la carte ? ...
Qu'utilises-tu pour compiler ?
La procédure pour compiler et flasher grbl-Mega-5X a l'aide du logiciel Arduino est la même que le Grbl d'origine, elle est décrite dans le Wiki de la page principale de Grbl ici :  
https://github.com/gnea/grbl/wiki/Compiling-Grbl

@++;
Gauthier.
